src/debug/dart_debug_impl.ts

```diff
@@ -234,6 +260,7 @@ export class DartDebugSession extends DebugSession {
 	): void {
 		if (this.childProcess != null)
 			this.childProcess.kill();
+		// TODO: Restart any paused threads for the attach case.
```

Comment:
Possibly this is as simple as doing what's in `continueRequest`. Do we also need to be sure to close the connection to Observatory? Normally we kill the process so it closes automatically, but I'm not sure here - we don't want to keep getting events after we disconnected (note: since a few versions ago, we run our debug adapter in-process in Code, so if we leak anything it'll hang around).

Reply:
Added the close.

I still need to look at continuing the threads (continueRequest takes a thread ID, so this will need to figure out which thread(s) are paused, which in my initial scan I didn't see currently stored internally; it looks like maybe up until now only VS Code itself has needed that information).

I also need to remove all breakpoints. Or at least, probably--it's a bit unusual here since breakpoints can come either from the debugger or from the VM itself, so unlike, say, a C++ debugger there's a question of what should happen if you disconnect from a process that had breakpoints set before you attached to it. My suspicion is that removing all breakpoints is probably the least surprising approach.